Tests/iaas/flavor-naming/flavor-add-extra-specs.py

```diff
@@ -0,0 +1,344 @@
+        chkval2 = real
+    if chkval < claim:
+        print(f"ERROR: Flavor {flvnm} claims {claim} {valnm}{extra}, but only has {real}. Needs fixing.",
+              file=sys.stderr)
+        return False
+    if chkval2 > claim:
+        print(f"WARNING: Flavor {flvnm} claims {claim} {valnm}{extra}, but overdelivers with {real}.",
+              file=sys.stderr)
+    return True
+
+
+def check_std_props(flavor, flvnm, extra=""):
+    """Check consistency of openstack props with parsed SCS name specs
+    Return no of errors found."""
+    errors = 0
+    #  vcpus
+    if not min_max_check(flavor.vcpus, flvnm.cpuram.cpus, "CPUs", flavor.name, extra):
+        errors += 1
+    #  ram
+    if not min_max_check(flavor.ram, flvnm.cpuram.ram*1024, "MiB RAM", flavor.name, extra):
+        errors += 1
+    #  disk
+    disksz = 0
+    if flvnm.disk:
+        disksz = flvnm.disk.disksize
+    if not min_max_check(flavor.disk, disksz, "GiB Disk", flavor.name, extra):
+        errors += 1
+    return errors
+
+
+def generate_name_v2(flavor, cpu_type, disk0_type):
+    """Generate an SCS- v2 name for flavor,
+    using cpu_type (and disk0_type if needed).
+    Returns string."""
+    cpuram = Main()
+    cpuram.cpus = flavor.vcpus
+    cpuram.cputype = cpu_type
+    cpuram.ram = int((flavor.ram+12)/512)/2.0
+    if flavor.disk:
+        disk = Disk()
+        disk.disksize = flavor.disk
+        disk.disktype = disk0_type
+        flv = Flavorname(cpuram, disk)
+    else:
+        flv = Flavorname(cpuram)
+    return outname(flv)
+
+
+def check_name_extra(flavor, ver, match, flname):
+    """Check for existence and consistency of scs names in extra specs
+    This assumes that a v1 or v2 name is used as main flavor name and should
+    match. If match is not set an v1->v2 or v2->v1 translation is needed.
+    ver needs to be set to 'v1' or 'v2'/'v3'/'v4'
+    flname is the SCS flavor name (may have been generated)
+    Returns non-zero if we need to perform an API call to perform a change.
+    """
+    spec = f"scs:name-{ver}"
+    errs = 0
+    need_name_set = True
+    if spec in flavor.extra_specs:
+        # Get name from scs:name-vN property
+        name = flavor.extra_specs[spec]
+        # If match=True has been passed, we expect it to match flname
+        # produces a warning on mismatch
+        if match and name != flname:
+            print(f"WARNING: {spec} {name} != flavor name {flname}",
+                  file=sys.stderr)
+        # Existing names must be parseable SCS names, check
+        # If we can parse the name AND there is no inconsistency
+        # we don't need to update the name.
+        try:
+            parsed = parser_v2(name)
+        except ValueError as exc:
+            try:
+                parsed = parser_v1(name)
+            except ValueError as exc2:
+                print(f"ERROR parsing {spec} {name}: {exc!r} {exc2!r}",
+                      file=sys.stderr)
+                need_name_set = False
```

Comment:
I think it might be worthwile to move the class `ParsingStrategy` from [cli.py](https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/flavor-naming/cli.py#L15) to flavor_names.py and to use it here as `ParsingStrategy(parsers=(parser_v2, parser_v1, ))`.

Reply:
OK, sound fine. I was unaware of its existence :-O

Reply:
I guess I would even change the class a little bit... We could add an attribute `vstr` and change the `parse` to `__call__` so that it has the same protocol as the class `Parser` from `flavor_names.py`. This would constitute a kind of _Composite pattern_ then ;)